src/esp/metadata/managers/StageAttributesManager.cpp

```diff
@@ -191,27 +191,51 @@ StageAttributes::ptr StageAttributesManager::initNewObjectInternal(
 
-    // set defaults for navmesh default handles and semantic mesh default
-    // handles
-    std::string navmeshFilename =
-        io::changeExtension(attributesHandle, ".navmesh");
-
-    if (Corrade::Utility::Directory::exists(navmeshFilename)) {
-      newAttributes->setNavmeshAssetHandle(navmeshFilename);
-    }
-    // Build default semantic descriptor file name
-    std::string houseFilename = io::changeExtension(attributesHandle, ".house");
-
-    if (!Corrade::Utility::Directory::exists(houseFilename)) {
-      houseFilename = io::changeExtension(attributesHandle, ".scn");
-    }
-    newAttributes->setHouseFilename(houseFilename);
-
-    // Build default semantic mesh file name
-    const std::string semanticMeshFilename =
-        Cr::Utility::Directory::splitExtension(houseFilename).first +
-        "_semantic.ply";
-    newAttributes->setSemanticAssetHandle(semanticMeshFilename);
+    if (attributesHandle != "NONE") {
+      // TODO when all datasets have configuration support, get rid of all these
+      // default settings
+
+      // set defaults for navmesh, semantic mesh and lexicon handles
+      // start with root stage name, including path but without final extension
+      // default navmesh should have .navmesh ext
+      std::string navmeshFilename =
+          this->findFilenameUsingCriteria(attributesHandle, {".navmesh"});
+      // if present, file was found, so set value (overriding attributes
+      // default)
+      if (!navmeshFilename.empty()) {
+        newAttributes->setNavmeshAssetHandle(navmeshFilename);
+      }
+
+      // Build default semantic descriptor file name, using extensions of
+      std::string ssdFileName = this->findFilenameUsingCriteria(
+          attributesHandle, {".house", ".scn", "_semantic.txt"});
+
+      // if not present, set hacky defaults for back compat expectations.
+      if (ssdFileName.empty()) {
+        if (attributesHandle.find("/replica_dataset") != std::string::npos) {
+          // replica hack until dataset gets appropriate configuration support
+          ssdFileName = Cr::Utility::Directory::join(
+              newAttributes->getFileDirectory(), "info_semantic.json");
+        } else {
+          // hack to support back-compat until configs are implemented for all
+          // datasets
+          ssdFileName =
+              Cr::Utility::Directory::splitExtension(attributesHandle).first +
+              ".scn";
+        }
+      }
+      newAttributes->setSemanticDescriptorFilename(ssdFileName);
+
+      // Build default semantic mesh filename as root stage name ending with
+      // "_semantic.ply", for back-compat with Mp3d
+      std::string semanticMeshFilename =
+          this->findFilenameUsingCriteria(attributesHandle, {"_semantic.ply"});
+      // if present, file was found, so set value (overriding attributes
+      // default)
+      if (!semanticMeshFilename.empty()) {
+        newAttributes->setSemanticAssetHandle(semanticMeshFilename);
```

Comment:
```suggestion
        newAttributes->setSemanticAssetHandle(std::move(semanticMeshFilename));
```

Reply:
All these string setters are const string& setters, so moving isn't really useful here, is it?